actionpack/test/controller/url_rewriter_test.rb

```diff
@@ -1,92 +0,0 @@
-# frozen_string_literal: true
-
-require "abstract_unit"
-require "controller/fake_controllers"
-
-class UrlRewriterTests < ActionController::TestCase
-  class Rewriter
-    def initialize(request)
-      @options = {
-        host: request.host_with_port,
-        protocol: request.protocol
-      }
-    end
-
-    def rewrite(routes, options)
-      routes.url_for(@options.merge(options))
-    end
-  end
-
-  def setup
-    @params = {}
-    @rewriter = Rewriter.new(@request)
-    @routes = ActionDispatch::Routing::RouteSet.new.tap do |r|
-      r.draw do
-        ActionDispatch.deprecator.silence do
-          get ":controller(/:action(/:id))"
-        end
-      end
-    end
-  end
-
-  def test_port
-    assert_equal("http://test.host:1271/c/a/i",
-      @rewriter.rewrite(@routes, controller: "c", action: "a", id: "i", port: 1271)
-    )
-  end
-
-  def test_protocol_with_and_without_separator
-    assert_equal("https://test.host/c/a/i",
-      @rewriter.rewrite(@routes, protocol: "https", controller: "c", action: "a", id: "i")
-    )
-
-    assert_equal("https://test.host/c/a/i",
-      @rewriter.rewrite(@routes, protocol: "https://", controller: "c", action: "a", id: "i")
-    )
-  end
-
-  def test_user_name_and_password
-    assert_equal(
-      "http://david:secret@test.host/c/a/i",
-      @rewriter.rewrite(@routes, user: "david", password: "secret", controller: "c", action: "a", id: "i")
-    )
-  end
-
-  def test_user_name_and_password_with_escape_codes
-    assert_equal(
-      "http://openid.aol.com%2Fnextangler:one+two%3F@test.host/c/a/i",
-      @rewriter.rewrite(@routes, user: "openid.aol.com/nextangler", password: "one two?", controller: "c", action: "a", id: "i")
-    )
-  end
-
-  def test_anchor
-    assert_equal(
-      "http://test.host/c/a/i#anchor",
-      @rewriter.rewrite(@routes, controller: "c", action: "a", id: "i", anchor: "anchor")
-    )
-  end
-
-  def test_anchor_should_call_to_param
-    assert_equal(
-      "http://test.host/c/a/i#anchor",
-      @rewriter.rewrite(@routes, controller: "c", action: "a", id: "i", anchor: Struct.new(:to_param).new("anchor"))
-    )
-  end
-
-  def test_anchor_should_be_uri_escaped
-    assert_equal(
-      "http://test.host/c/a/i#anc/hor",
-      @rewriter.rewrite(@routes, controller: "c", action: "a", id: "i", anchor: Struct.new(:to_param).new("anc/hor"))
-    )
-  end
```

Comment:
I didn't move this test. Maybe I should? The name of the test doesn't match what the test is doing. 
Nothing is escaped here. 

We do have a test for escaping (or not escaping) [here](https://github.com/rails/rails/blob/48d4e6e02b9b44553118f7ea955bd3b48a0512da/actionpack/test/controller/url_for_test.rb#L85-L95)